### PR TITLE
Runtime 23.08

### LINKED ISSUE #6: Update to runtime 21.08
Should have been updated a while ago

### PR DESCRIPTION
**Pull request by bbhtt**
Closes https://github.com/flathub/com.scoutshonour.dtipbijays/issues/6
Closes https://github.com/flathub/com.scoutshonour.dtipbijays/pull/8

pygame was switched to the one from shared-modules because I just fixed couple of build errors https://github.com/flathub/shared-modules/commit/b5eb7c51ed89abee30e703f5037d92abd2ceee53.  No point in duplicating pygame manifest (and effort) when it can be maintained in a single place.

Renpy build fails  due to ffmpeg deprecations and removals (it looks for libavcodec headers in runtime). So it necessitates building our own copy of ffmpeg 4.2.2 in `/app`.

Rest should be self-explanatory.

### CODE REVIEW

**Comment by bbhtt** [comment]
I'll add some cleanups once https://github.com/flathub/shared-modules/pull/263 is merged.

File: com.scoutshonour.dtipbijays.yaml [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-    version: "20.08"
+    version: "23.08"
     autodelete: false
 finish-args:
   - --socket=x11
   - --share=ipc
   - --device=dri
   - --socket=pulseaudio
-  - --filesystem=~/.renpy:create
+  - --persist=.renpy
```
Comment (MateusRodCosta) [inline com.scoutshonour.dtipbijays.yaml]: I take a bit of issue with this change.

So, basically, [dtipijays](https://flathub.org/apps/com.scoutshonour.dtipbijays) is somewhat of a sequel to [Digital](https://flathub.org/apps/com.scoutshonour.Digital). Both of them use a Ren'py feature called [Multi-Game Persistence](https://www.renpy.org/doc/html/persistent.html#multi-game-persistence).
This means that basically both games can share some data with each other. In this case, Digital saves some data when you play it (such as first name and last name, which you have to type when starting the game) and dtipijays accesses it (a certain scene can show the name you inputted in the last game as a joke).

For both games that shared persistent data seems to be stored in `~/.renpy/persistent/scoutshonour.com`.
So, I assume, to have this feature working you would need to have at least ro access to that file, but preferably rw access (I don't know for sure if dtipijays saves to it, I only know for sure that Digital does).

That Ren'py feature, btw, is the reason it was set to `--filesystem=~/.renpy:create`, so I am fine with changing as long as the feature can be kept working.

**Review by MateusRodCosta — COMMENTED** [review]
Honestly this seems to mostly work fine from a quick testing.

I only request you take note of my comment on the .renpy folder.

File: com.scoutshonour.dtipbijays.yaml (hunk 1 above)
Comment (bbhtt) [reply com.scoutshonour.dtipbijays.yaml]: I'd hope that it works but i can't really check, right now, sorry.

I'm fine with reverting that change for now.

Feel free to check it later and introduce the fixes if needed.

It's not too much of an issue.

Comment (bbhtt) [reply com.scoutshonour.dtipbijays.yaml]: Dropped, this won't work unless you give the other flatpak permission to the per app sandbox folder.

persist creates the directory inside the per app sandbox and bind mounts to real home where the app expects it, but this wouldn't be shareable.

**Review by MateusRodCosta — APPROVED** [review]
LGTM.

I believe this is good enough for the runtime changes, thanks a lot!

We can discuss later on how to deal with the multi-game persistence feature.

I also have to make some changes to the AppData later, apparently the version we ship is 1.1 (that's what shows up in the tarball we download) but the AppData has 1.0 information which is replicated for the flatpak information.

**Pull request merged**